test/extensions/filters/http/adaptive_concurrency/adaptive_concurrency_filter_integration_test.h

```diff
@@ -0,0 +1,100 @@
+#include "test/integration/http_integration.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+const std::string kAdaptiveConcurrencyFilterConfig =
+    R"EOF(
+name: envoy.filters.http.adaptive_concurrency
+config:
+  gradient_controller_config:
+    sample_aggregate_percentile:
+      value: 50
+    concurrency_limit_params:
+      concurrency_update_interval:
+        nanos: 100000000 # 100ms
+    min_rtt_calc_params:
+      interval:
+        seconds: 30
+      request_count: 50
+)EOF";
+
+const std::string kFaultFilterConfig =
+    R"EOF(
+name: envoy.fault
+config:
+    delay:
+        header_delay: {}
+        percentage:
+            numerator: 100
+            denominator: HUNDRED
+  )EOF";
+
+const std::string kConcurrencyLimitGaugeName =
+    "http.config_test.adaptive_concurrency.gradient_controller.concurrency_limit";
+const std::string kRequestBlockCounterName =
+    "http.config_test.adaptive_concurrency.gradient_controller.rq_blocked";
+const std::string kMinRTTGaugeName =
+    "http.config_test.adaptive_concurrency.gradient_controller.min_rtt_msecs";
+
+// The default delay introduced to each sent request.
+const uint32_t kDefaultRequestDelayMs = 50;
+
+class AdaptiveConcurrencyIntegrationTest
+    : public HttpIntegrationTest,
+      public testing::TestWithParam<Network::Address::IpVersion> {
+public:
+  AdaptiveConcurrencyIntegrationTest()
+      : HttpIntegrationTest(Http::CodecClient::Type::HTTP2, GetParam()) {}
+
+  void initializeFilter() {
+    // We use the fault filter (for delays) after the adaptive concurrency filter to introduce a
+    // "service latency" to the test. This way, time is moved forward with each response.
```

Comment:
Why can't you just use simulated time and just advance time before upstream responds?

Wherever we end up on this, please verify that you have done a de-flaking run on these new tests (`--runs_per_test=1000`) to verify a lack of flakes.